demo/templates/demo/includes/streamfield.html

```diff
@@ -0,0 +1,46 @@
+{% load wagtailcore_tags wagtailimages_tags %}
+
+<div class="stream-field">
+    {% for child in content %}
+        {% if child.block_type == 'h2' %}
+            <h2>{{ child }}</h2>
+        {% elif child.block_type == 'h3' %}
+            <h3>{{ child }}</h3>
+        {% elif child.block_type == 'h4' %}
+            <h4>{{ child }}</h4>
+        {% elif child.block_type == 'intro' %}
+            <div class="intro">{{ child }}</div>
+        {% elif child.block_type == 'aligned_html' %}
+            {% if child.value.alignment == 'normal' %}
+                {{ child.value.bound_blocks.html.render }}
+            {% else %}
+                {{ child.value.bound_blocks.html.render }}
+            {% endif %}
+        {% elif child.block_type == 'pullquote' %}
+            <blockquote>
+                {{ child.value.quote }}
+                {% if child.value.attribution %}<span>- {{ child.value.attribution }}</span>{% endif %}
+            </blockquote>
+        {% elif child.block_type == 'paragraph' %}
+            {{ child.value|richtext }}
+        {% elif child.block_type == 'aligned_image' %}
+            <div class="img-wrapper {{ child.value.alignment }}">
+                {% if child.value.alignment == "left" or child.value.alignment == "right" %}
+                    {% image child.value.image width-300 as theimage %}
+                {% else %}
+                    {% image child.value.image width-1280 as theimage %}
+                {% endif %}
+
+                <img src="{{ theimage.url }}" width="{{ theimage.width }}" height="{{ theimage.height }}" alt="{{ theimage.alt }}" />
+
+                {% if child.value.caption %}
+                    <div class="caption">
+                        {{ child.value.caption|richtext }}
+                    </div>
+                {% endif %}
+            </div>
+        {% else %}
+            {{ child }}
+        {% endif %}
+    {% endfor %}
+</div>
```

Comment:
We'd really need to get into code profiling to know what the difference in performance is, and whether it's significant enough to choose the faster version over the neater code.

There are some places in Wagtail core (e.g. `render` methods on blocks) where I went with the 'received wisdom' that templates are a lot slower than manual string-munging in Python code - perhaps that's what you're thinking of. Admittedly, that *is* a place where ideally we would do some profiling. However, I'd argue that that's a bit of a different case to this - firstly, it's about template code vs non template code, rather than two different styles of template code; and secondly, I think it makes sense for Wagtail core code to favour performance over readability more than a typical Wagtail project would.